presentation/src/main/java/com/architect/coders/mu8/ui/activity/ComicsActivity.kt

```diff
@@ -0,0 +1,76 @@
+package com.architect.coders.mu8.ui.activity
+
+import android.os.Bundle
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.architect.coders.mu8.R
+import com.architect.coders.mu8.data.remote.repository.ComicRepository
+import com.architect.coders.mu8.ui.adapter.ComicsAdapter
+import com.architect.coders.mu8.ui.viewmodel.ComicsViewModel
+import com.architect.coders.mu8.ui.viewmodel.ComicsViewModel.UiModel
+import com.architect.coders.mu8.ui.viewmodel.ComicsViewModel.UiModel.*
+import com.architect.coders.mu8.utils.getViewModel
+import com.architect.coders.mu8.utils.makeItGone
+import com.architect.coders.mu8.utils.makeItVisible
+import com.architect.codes.mu8.interactor.comics.ComicsUseCase
+import com.google.android.material.snackbar.Snackbar
+
+class ComicsActivity : AppCompatActivity() {
+
+    private val progress: ProgressBar by lazy { findViewById<ProgressBar>(R.id.comic_activity_progress) }
+    private val recycler: RecyclerView by lazy { findViewById<RecyclerView>(R.id.comic_activity_recycler) }
+
+    private lateinit var viewModel: ComicsViewModel
+    private lateinit var adapter: ComicsAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_comics)
+
+        viewModel = getViewModel {
+            ComicsViewModel(
+                ComicsUseCase(
+                    ComicRepository()
+                )
+            )
+        }
+
+        adapter = ComicsAdapter(viewModel::onComicClicked)
+        recycler.layoutManager = GridLayoutManager(this, 2)
```

Comment:
What is `2`? Please create a meaningful constant name for this.

Reply:
done grid colums